crates/astria-composer/src/executor/builder.rs

```diff
@@ -0,0 +1,75 @@
+use std::time::Duration;
+
+use astria_core::sequencer::v1::{
+    transaction::action::SequenceAction,
+    Address,
+};
+use astria_eyre::{
+    eyre,
+    eyre::{
+        eyre,
+        Context,
+    },
+};
+use ed25519_consensus::SigningKey;
+use ethers::core::k256::elliptic_curve::zeroize::Zeroize;
+use secrecy::{
+    ExposeSecret,
+    SecretString,
+};
+use tokio::sync::watch;
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    executor,
+    executor::Status,
+};
+
+pub(crate) struct Builder {
+    pub(crate) sequencer_url: String,
+    pub(crate) private_key: SecretString,
```

Comment:
We were initially passing them as references but I chose to avoid references here so that we don't have to deal with lifetime related stuff. Also String and SecretString should be fairly lightweight to clone.